.github/workflows/java-compatibility.yml

```diff
@@ -0,0 +1,103 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+
+# See https://docs.gradle.org/current/userguide/compatibility.html
+name: Gradle Java Compatibility Tests
+on: [ pull_request ]
+jobs:
+  java17:
+    name: "Compatibility: java ${{ matrix.java }}/gradle ${{ matrix.gradle }}"
+    runs-on: "ubuntu-latest"
+    strategy:
+      matrix:
+        avro: [ "1.12.0" ]
+        gradle: [ "8.14" ]
+        java: [ "17" ]
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
+        with:
+          distribution: "zulu"
+          java-version: ${{ matrix.java }}
+      - uses: gradle/gradle-build-action@v2
+        with:
+          build-root-directory: lang/java/gradle-plugin
+          gradle-executable: lang/java/gradle-plugin/gradlew
+          arguments: testCompatibility -PavroVersion=${{ matrix.avro }} -PgradleVersion=${{ matrix.gradle }}
+  java18:
+    name: "Compatibility: java ${{ matrix.java }}/gradle ${{ matrix.gradle }}"
+    runs-on: "ubuntu-latest"
+    strategy:
+      matrix:
+        avro: [ "1.12.0" ]
+        gradle: [ "8.14" ]
+        java: [ "18" ]
```

Comment:
I'd suggest to not test non LTS Java versions
Java 18, 19, 20, 22 are no more supported anyway (unless you pay to someone)